src/ProjectTemplates/Shared/AspNetProcess.cs

```diff
@@ -134,13 +136,13 @@ public async Task AssertPagesOk(IEnumerable<Page> pages)
 
         public async Task ContainsLinks(Page page)
         {
-            var response = await RequestWithRetries(client =>
+            var response = await RetryHelper.RetryRequest(async () =>
             {
                 var request = new HttpRequestMessage(
                     HttpMethod.Get,
                     new Uri(ListeningUri, page.Url));
-                return client.SendAsync(request);
-            }, _httpClient);
+                return await _httpClient.SendAsync(request);
+            }, logger: NullLogger.Instance);
```

Comment:
Does this prevent logging of errors hit in the final attempt when that also fails?

Reply:
Nope, the error/exception will show up at a higher level. The other helper didn't have logging there, so we are not loosing anything here. Same for all the other changes.